### ISSUE TITLE
Address usage of deprecated 'imp' module.

### ISSUE THREAD

**Issue by danielballan** [intake/intake#477]
This usage is deprecated and generates warnings in the tests.

https://github.com/intake/intake/blob/3678714200774efe0bd725b8d8b00f4743ed8dd7/intake/source/discovery.py#L301-L305

A downside of this approach is it bypasses the import machinery's normal caching mechanism, such that if the same plugin is loaded multiple times by intake, each load produces a new `type` instance even though they represent the same Python file and should be equivalent. (Maybe this is why `imp` is deprecated and that method is now un-documented. I'm not up to speed on Python import internals.)

We could look for a cleaner way to do this, or evaluate whether we want to do it at all.

**Comment by danielballan** [comment]
Referring to #478, this:

```py
register_driver('thing', A)
register_driver('thing', A)  # OK, this is redundant but harmless
```

only works if `A` and `A` are literally the same type. `imp.load_source` causes trouble because it exec's the same file twice and produces distinct types, similar to this issue:

```py
# example.py
class A:
    ...
```

```py
In [1]: import example                                                                                                                                                                        

In [2]: a = example.A()                                                                                                                                                                       

In [3]: isinstance(a, example.A)                                                                                                                                                              
Out[3]: True

In [4]: from imp import reload                                                                                                                                                                
/home/dallan/venv/bnl/bin/ipython:1: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  #!/home/dallan/venv/bnl/bin/python3

In [5]: reload(example)                                                                                                                                                                       
Out[5]: <module 'example' from '/home/dallan/Repos/bnl/intake/example.py'>

In [6]: isinstance(a, example.A)                                                                                                                                                              
Out[6]: False
```

**Comment by danielballan** [comment]
(Fully addressed by #478, for the record.)
